core/src/main/java/com/graphhopper/routing/weighting/custom/PriorityCustomConfig.java

```diff
@@ -0,0 +1,170 @@
+import org.locationtech.jts.geom.prep.PreparedGeometryFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+final class PriorityCustomConfig {
+    private List<EdgeToValueEntry> priorityList = new ArrayList<>();
+
+    public PriorityCustomConfig(CustomModel customModel, EncodedValueLookup lookup) {
+        add(lookup, customModel.getVehicleWeight(), "vehicle_weight", MaxWeight.KEY);
+        add(lookup, customModel.getVehicleWidth(), "vehicle_width", MaxWidth.KEY);
+        add(lookup, customModel.getVehicleHeight(), "vehicle_height", MaxHeight.KEY);
+        add(lookup, customModel.getVehicleLength(), "vehicle_length", MaxLength.KEY);
+
+        for (Map.Entry<String, Object> entry : customModel.getPriority().entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+
+            if (value == null) {
+                throw new IllegalArgumentException("Missing value for " + key + " in 'priority'");
+            } else if (value instanceof Number) {
+                if (key.startsWith(GeoToValueEntry.key(""))) {
+                    Geometry geometry = GeoToValueEntry.pickGeometry(customModel, key);
+                    priorityList.add(new GeoToValueEntry(new PreparedGeometryFactory().create(geometry), ((Number) value).doubleValue(), 1));
+                } else {
+                    BooleanEncodedValue encodedValue = getEV(lookup, "priority", key, BooleanEncodedValue.class);
+                    priorityList.add(new BooleanToValueEntry(encodedValue, ((Number) value).doubleValue(), 1));
+                }
+            } else if (value instanceof Map) {
+                EnumEncodedValue enumEncodedValue = getEV(lookup, "priority", key, EnumEncodedValue.class);
+                Enum[] enumValues = enumEncodedValue.getValues();
+                double[] values = createEnumToDoubleArray("priority." + key, 1, 0, 100,
+                        enumValues, (Map<String, Object>) value);
+                normalizeFactor(values, 1);
+                priorityList.add(new EnumToValueEntry(enumEncodedValue, values));
+            } else {
+                throw new IllegalArgumentException("Type " + value.getClass() + " is not supported for " + key + " in 'priority'");
+            }
+        }
+    }
+
+    static <T extends EncodedValue> T getEV(EncodedValueLookup lookup, String name, String key, Class<T> encodedValueType) {
+        if (!lookup.hasEncodedValue(key))
+            throw new IllegalArgumentException("Cannot find '" + key + "' specified in '" + name + "'");
+        T ev = lookup.getEncodedValue(key, encodedValueType);
+        if (!encodedValueType.isAssignableFrom(ev.getClass()))
+            throw new IllegalArgumentException("Expected class '" + encodedValueType.getSimpleName() + "' for '" + key
+                    + "' specified in '" + name + "'. But was " + ev.getClass().getSimpleName());
```

Comment:
Also here I think we should improve the error message. This error happens if we use some enum encoded value with a plain number, like `road_class: 15` or a boolean encoded value with a map, like `get_off_bike: { motorway: 0.3}`? 

```
So instead of `Expected class 'BooleanEncodedValue' for 'road_class' specified in 'speed_factor'. But was EnumEncodedValue
```

Maybe something like

```
The encoded value 'road_class' is of type EnumEncodedValue, its value must be a map (not a number) in 'speed_factor'.
```

Reply:
Yes, please go ahead. Error messages can be certainly improved :+1: